gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +89,101 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
     return ht
 
 
+def create_pangolin_grch38_ht(vcf_path: str) -> hl.Table:
+    """
+    Create a Hail Table with Pangolin score for splicing for GRCh38.
+
+    .. Note::
+    The Pangolin scores were generated for variants in gene body only. The first run contains variants from gnomAD v3 genomes.
+
+    :param vcf_path: path to the VCF files with Pangolin scores for splicing.
+    :return: Hail Table with Pangolin score for splicing for GRCh38.
+    """
+    mt = hl.import_vcf(vcf_path, min_partitions=1000, reference_genome="GRCh38")
+
+    ht = mt.rows()
+    logger.info("Number of rows in original Pangolin Hail Table: %s", ht.count())
+
+    logger.info("Exploding Pangolin scores...")
+    # `explode` will eliminate rows with empty array
+    ht = ht.annotate(pango=ht.info.Pangolin[0].split(delim="\|\|"))
+    ht = ht.explode(ht.pango)
+    logger.info("Number of rows in exploded Pangolin Hail Table: %s", ht.count())
+
+    logger.info("Annotating Pangolin scores...")
+    ht = ht.annotate(
+        pangolin=hl.struct(
+            gene=ht.pango.split(delim=":|\\|")[0],
+            positions=(
+                hl.empty_array(hl.tint32).append(
+                    hl.int(ht.pango.split(delim=":|\\|")[1])
+                )
+            ).append(hl.int(ht.pango.split(delim=":|\\|")[3])),
+            delta_scores=(
+                hl.empty_array(hl.tfloat64).append(
+                    hl.float(ht.pango.split(delim=":|\\|")[2])
+                )
+            ).append(hl.abs(hl.float(ht.pango.split(delim=":|\\|")[4]))),
+        )
```

Comment:
A note clarifying why we are the scores are in two locations in the VCF would be helpful here. Also, you take a different approach with this tool than you did with splice_ai (where you defined variables before the annotate function and then used those variables within a hl.map call to cast them to floats within the annotate) . I think either works (though I found spliceai's approach a bit cleaner, but for readability sake, the approach should be kept consistent.